src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/MLKemImplementation.OpenSsl.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Security.Cryptography
+{
+    internal sealed class MLKemImplementation : MLKem
+    {
+        private SafeEvpPKeyHandle _key;
+
+        // OpenSSL is expected to give "all or none" support.
+        internal static new bool IsSupported => Interop.Crypto.EvpKemAlgs.MlKem512 is not null;
```

Comment:
My ML-DSA stub made a MLKemImplementation.cs that just defined everything as partial, so that errors are more clearly "this method is missing" rather than "overload resolution couldn't figure out what to call" (and it gives a nice thing to copy/paste from when setting up a new platform).  As part of that I ended up renaming IsSupported in the impl, either because `new partial` didn't work, or because it felt icky.

It doesn't look like you used the partial decl file for KEM.  I'm open to criticism that it doesn't add much and we shouldn't have them, but since it currently means ML-KEM is structured differently than ML-DSA it seems worth discussing.

Reply:
I have some mixed feelings on this. On one hand, yes the partial is a nice forcing function to make sure the statics are present (though the compiler will do that anyway, just with a slightly confusing error).

On the other hand, I value the lack of a partial. We use a lot of partials in the runtime and it can be hard to keep straight. I like "it's all one class in a file" and nice and tidy.

When partials bring value - like XML docs on public type - then I would absolutely agree that a partial is the right thing.

I don't mind adding it (too much) but its more lines of code and a file that doesn't _need_ to exist.